client/src/components/admin/problem.js

```diff
@@ -0,0 +1,101 @@
+import withStyles from '../../components/jss'
+import { useState, useCallback } from 'preact/hooks'
+
+import { updateChallenge, deleteChallenge } from '../../api/admin/challs'
+import { withToast } from '../../components/toast'
+
+const Problem = withStyles({
+  frame: {
+    marginBottom: '1em',
+    paddingBottom: '0.625em'
+  },
+  description: {
+    '& a': {
+      display: 'inline',
+      padding: 0
+    }
+  },
+  divider: {
+    margin: '0.625em',
+    width: '80%'
+  },
+  header: {
+    marginTop: '15px',
+    '& input': {
+      margin: '5px 0'
+    }
+  }
+}, ({ classes, problem, toast }) => {
+  const [flag, setFlag] = useState(problem.flag)
+  const handleFlagChange = useCallback(e => setFlag(e.target.value), [])
+
+  const [description, setDescription] = useState(problem.description)
+  const handleDescriptionChange = useCallback(e => setDescription(e.target.value), [])
+
+  const [category, setCategory] = useState(problem.category)
+  const handleCategoryChange = useCallback(e => setCategory(e.target.value), [])
+
+  const [author, setAuthor] = useState(problem.author)
+  const handleAuthorChange = useCallback(e => setAuthor(e.target.value), [])
+
+  const [name, setName] = useState(problem.name)
+  const handleNameChange = useCallback(e => setName(e.target.value), [])
+
+  const handleUpdate = e => {
```

Comment:
This should be in a `useCallback` hook to avoid re-registering a new event handler each render